lib/mock/mock-agent.js

```diff
@@ -11,31 +11,46 @@ const {
   kNetConnect,
   kGetNetConnect,
   kOptions,
-  kFactory
+  kFactory,
+  kMockAgentMockCallHistoryName,
+  kMockAgentRegisterCallHistory,
+  kMockAgentIsCallHistoryEnabled,
+  kMockAgentAddCallHistoryLog,
+  kMockCallHistoryGetByName,
+  kMockCallHistoryClearAll,
+  kMockCallHistoryDeleteAll,
+  kMockCallHistoryCreate,
+  kMockCallHistoryAddLog
 } = require('./mock-symbols')
 const MockClient = require('./mock-client')
 const MockPool = require('./mock-pool')
 const { matchValue, buildMockOptions } = require('./mock-utils')
 const { InvalidArgumentError, UndiciError } = require('../core/errors')
 const Dispatcher = require('../dispatcher/dispatcher')
 const PendingInterceptorsFormatter = require('./pending-interceptors-formatter')
+const { MockCallHistory } = require('./mock-call-history')
 
 class MockAgent extends Dispatcher {
   constructor (opts) {
     super(opts)
 
     this[kNetConnect] = true
     this[kIsMockActive] = true
+    this[kMockAgentIsCallHistoryEnabled] = Boolean(opts?.enableCallHistory)
```

Comment:
To avoid surprises, better to validate that the `opts.enableCallHistory` is a true `boolean` value, or throw otherwise.

Reply:
Done https://github.com/nodejs/undici/pull/4029/commits/7d9c6793f79ddf1231980823e47061a378a21107